docs/sources/trackers/javascript-trackers/web-tracker/tracking-events/element-tracking/index.md

```diff
@@ -0,0 +1,793 @@
+import { enableFormTracking } from '@snowplow/browser-plugin-form-tracking';
+
+// access a context generator aware of the startElementTracking "components" configuration
+const [componentGenerator, componentGeneratorWithDetail] = getComponentListGenerator();
+
+// this will attach the component_parents entity to events generated by these plugins that show the component hierarchy
+enableLinkClickTracking({ options: { ... }, psuedoClicks: true, context: [componentGenerator] });
+enableFormTracking({ context: [componentGenerator] });
+
+// componentGeneratorWithDetail will also populate element_detail entities for each component, but is not directly compatible with the above plugin APIs
+```
+  </TabItem>
+</Tabs>
+
+```json reference title="Entity - iglu:com.snowplowanalytics.snowplow/component_parents/jsonschema/1-0-0" referenceLinkText="See schema on Github"
+https://github.com/snowplow/iglu-central/blob/master/schemas/com.snowplowanalytics.snowplow/component_parents/jsonschema/1-0-0
+```
+
+#### Element contents
+
+The `element_content` schema is attached when you use the [`contents` setting](#element-data).
+
+There will likely be multiple instances of this entity in individual events, and the list of them are a flattened tree representation of the nested configuration provided.
+Each instance contains references to the parent `element_content` or `element` entity instance that contains it in the `parent_name` and `parent_index` properties (via `element_name` and `element_index`, respectively).
+Nested `details` configurations are also used to populate the `attributes` for each instance.
+
+```json reference title="Entity - iglu:com.snowplowanalytics.snowplow/element_content/jsonschema/1-0-0" referenceLinkText="See schema on Github"
+https://github.com/snowplow/iglu-central/blob/master/schemas/com.snowplowanalytics.snowplow/element_content/jsonschema/1-0-0
+```
+
+#### Custom context
+
+Custom entities can be attached to the events generated by the plugin.
+
+- You can include `context` alongside `elements` when calling `startElementTracking`. You can pass an array of static entities _or_ a callback function that returns such an array. The function will receive the element that the event is relevant to, and the matching rule that defined the event should fire.
+- Individual rules may also contain specific `context` in the same format as above.
+
+## Examples
+
+Here we'll show some example rules for simple use cases on the [Snowplow website](https://snowplow.io/) ([snapshot at time of writing](https://web.archive.org/web/20250422013533/https://snowplow.io/)).
+
+<details>
+  <summary>Scroll sections</summary>
+
+  The homepage has content grouped into distinct "layers" as you scroll down the page.
+  To see when users scroll down to each section, we can track an `expose` event for each section.
+  We can capture the header element text to identify each one.
+
+  ```json title="Rule configuration"
+  {
```

Comment:
I think it would be a bit more user friendly to give the full snippet including the `startElementTracking` JS call? Would take a bit less mental effort to understand and could let people copy stuff more easily.

Reply:
Good call, I tried to avoid this to not have to duplicate a bunch of `Tab` stuff for the different APIs. I'll just add a note mentioning the syntax for the examples.